drivers/sensor/sensor_shell.c

```diff
@@ -460,6 +521,118 @@ static int cmd_get_sensor_info(const struct shell *sh, size_t argc, char **argv)
 #endif
 }
 
+enum sample_stats_state {
+	SAMPLE_STATS_STATE_UNINITIALIZED = 0,
+	SAMPLE_STATS_STATE_ENABLED,
+	SAMPLE_STATS_STATE_DISABLED,
+};
+
+struct sample_stats {
+	int64_t accumulator;
+	uint32_t count;
+	uint64_t sample_window_start;
+	enum sample_stats_state state;
+};
+
+static void data_ready_trigger_handler(const struct device *sensor,
+				       const struct sensor_trigger *trigger)
+{
+	static struct sample_stats stats[SENSOR_CHAN_ALL];
+	const int64_t now = k_uptime_get();
+	struct sensor_value value;
+
+	if (sensor_sample_fetch(sensor)) {
+		LOG_ERR("Failed to fetch samples on data ready handler");
+	}
+	for (int i = 0; i < SENSOR_CHAN_ALL; ++i) {
+		int rc;
+
+		/* Skip disabled channels */
+		if (stats[i].state == SAMPLE_STATS_STATE_DISABLED) {
+			continue;
+		}
+		/* Skip 3 axis channels */
+		if (i == SENSOR_CHAN_ACCEL_XYZ || i == SENSOR_CHAN_GYRO_XYZ ||
+		    i == SENSOR_CHAN_MAGN_XYZ) {
+			continue;
+		}
+
+		rc = sensor_channel_get(sensor, i, &value);
+		if (rc == -ENOTSUP && stats[i].state == SAMPLE_STATS_STATE_UNINITIALIZED) {
+			/* Stop reading this channel if the driver told us it's not supported. */
+			stats[i].state = SAMPLE_STATS_STATE_DISABLED;
+		}
+		if (rc != 0) {
+			/* Skip on any error. */
+			continue;
+		}
+		/* Do something with the data */
+		stats[i].accumulator += value.val1 * INT64_C(1000000) + value.val2;
+		if (stats[i].count++ == 0) {
+			stats[i].sample_window_start = now;
+		} else if (now > stats[i].sample_window_start + CONFIG_SAMPLE_PRINT_TIMEOUT_MS) {
```

Comment:
`CONFIG_SAMPLE_PRINT_TIMEOUT_MS` was defined in the Kconfig of the sample, needs to be moved to `drivers/sensor/Kconfig` but probably better yet, we might want to move the sensor_shell configs into their own Kconfig so things don't get messy, maybe add a new `drivers/sensor/Kconfig.sensor_shell` and include it from the other one?

Reply:
Relocated Kconfig to `drivers/sensor/Kconfig`